### PR TITLE
switch strimzi from zookeeper to kraft

### PR DESCRIPTION
**Pull request by maschmid**
- :gift: Use kraft for strimzi by default

to get rid of flakiness in zookeeper startup

### CODE REVIEW

File: hack/lib/strimzi.bash [hunk 1]
```diff
@@ -20,89 +20,92 @@ function install_strimzi_operator {
...
+metadata:
+  name: kafka
+  namespace: kafka
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - controller
+    - broker
```
Comment (pierDipi) [inline hack/lib/strimzi.bash:94]: With this configuration, the 3 replicas will be both controllers and brokers, right?

File: hack/lib/strimzi.bash [hunk 2]
```diff
@@ -20,89 +20,92 @@ function install_strimzi_operator {
...
+  storage:
+    type: jbod
+    volumes:
+    - id: 0
+      type: persistent-claim
+      size: 100Gi
+      deleteClaim: false
```
Comment (pierDipi) [inline hack/lib/strimzi.bash:101]: I see we use `false` just like the previous one, but what's the reason ?

Comment (maschmid) [reply hack/lib/strimzi.bash:101]: We've inherited that from https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/kafka/kafka-persistent.yaml , it defaults to `false` , so I assume it's just there in the example to show that it exists.

It doesn't really matter, we don't delete Kafka in CI, and the PVC (and its PV) would be deleted with the cluster anyway

**Comment by creydr** [comment]
@maschmid @pierDipi do you want to have this in, before we cut 1.33?

`error=failed to load configuration: got unexpected http 404 status code from configresolver: failed to get config: could not find any config for branch main on repo openshift-knative/serverless-operator`

/retest

**Comment by matzew** [comment]
I'd _not_ block the release for this - but its good to move towards this setup, in the next weeks. IMO

**Comment by maschmid** [comment]
It doesn't block 1.33 , but feel free to merge  it right away.

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**